browser/playlist/test/playlist_browsertest.cc

```diff
@@ -333,3 +342,85 @@ IN_PROC_BROWSER_TEST_F(PlaylistBrowserTest, PlaylistTabHelper) {
   WaitUntil(base::BindLambdaForTesting(
       [&]() { return playlist_tab_helper->found_items().size() == 0; }));
 }
+
+class PlaylistBrowserTestWithSitesUsingMediaSource
+    : public PlaylistBrowserTest {
+ public:
+  // PlaylistBrowserTest:
+  void SetUpHTTPSServer() override {
+    https_server_ = std::make_unique<net::EmbeddedTestServer>(
+        net::test_server::EmbeddedTestServer::TYPE_HTTPS);
+    https_server_->RegisterRequestHandler(base::BindRepeating(
+        &PlaylistBrowserTestWithSitesUsingMediaSource::Serve,
+        https_server_.get()));
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+ private:
+  static std::unique_ptr<net::test_server::HttpResponse> Serve(
+      net::EmbeddedTestServer* server,
+      const net::test_server::HttpRequest& request) {
+    GURL absolute_url = server->GetURL(request.relative_url);
+    auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+    response->set_code(net::HTTP_OK);
+    response->set_content(R"html(
+        <html>
+        <meta property="og:image" content="/img.jpg">
+        <body>
+          <video id="vid"/>
+        </body>
+        <script>
+          if (window.MediaSource) {
+            const videoElement = document.querySelector('#vid');
+            videoElement.src = URL.createObjectURL(new MediaSource());
+          } else {
+            const videoElement = document.querySelector('#vid');
+            videoElement.src = '/test.mp4';
+          }
+        </script>
+        </html>
+      )html");
+    response->set_content_type("text/html; charset=utf-8");
+    return response;
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(PlaylistBrowserTestWithSitesUsingMediaSource,
```

Comment:
this is a lot to pack into a single test and a single test is not sufficient for this added functionality. For instance you have only tested one case for `ShouldRefetchMediaSourceToCache`. I have mentioned before that testing a single correct case is not adequate testing. What if there are 0 itmes? What happens if it is called again and all items are already cached? Some items are cached? What happens if the fetch fails or it fails to write to disk? You need to cover edge cases, errors, etc...

Reply:
Good point. but I think testing caching part is a little bit out of this PR's scope. Let me handle that in a follow-up. I added more tests to cover this change.